taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
-                                                onChange = {event => {
-                                                    changeData(lang, index, event.target.value)
-                                                }}
-                                                value={tag} 
-                                                variant="outlined" />
-                                        </Paper>
+                                        <Stack key={index} direction="row" alignItems="center">
+                                            <Paper component={Stack} direction="column" sx={{ml: 4, width: 200}}>
+                                                <TextField 
+                                                    size="small" 
+                                                    sx={{mt: 1}} 
+                                                    onChange = {event => {
+                                                        changeData(lang, index, event.target.value)
+                                                    }}
+                                                    value={tag} 
+                                                    variant="outlined" />
+                                            </Paper>
+                                            <IconButton sx={{ml: 1, mt: 1, color: "#808080"}} onClick={() => handleDelete(lang, index)}>
+                                                 <DeleteOutlineIcon />
+                                            </IconButton>
+                                        </Stack>
                                     )
                                 })
                             }
                         </Box>
                     )
                 } )
             }
+            {/* Dialog box for adding translations */}
+            <Dialog open={open} onClose={handleClose}>
+                <DialogTitle>Add a language</DialogTitle>
+                <DialogContent>
+                <DialogContentText>
+                    Enter the two letter language code for the language to be added.
+                </DialogContentText>
+                <TextField
+                    autoFocus
+                    margin="dense"
+                    onKeyPress={(e) => { (e.key === 'Enter') && isValidLanguageCode && handleAddTranslation(newLanguageCode, e) }} 
+                    onChange={(e) => { 
+                        setNewLanguageCode(e.target.value);
+                        const validateBool = ISO6391.validate(e.target.value);
+                        const ifDuplicateBool = renderedTranslations.some(el => (el.languageCode === e.target.value)) || 
+                                                nodeObject.main_language === e.target.value
+                        validateBool && !ifDuplicateBool ? setisValidLanguageCode(true) : setisValidLanguageCode(false)
+                        validateBool && !ifDuplicateBool ? setBtnDisabled(false) : setBtnDisabled(true)
```

Comment:
this could be a function?

Reply:
also you are using ternaries in a non standar  way, those are mainly for returning something.
 `if (validateBool && !ifDuplicateBool ) {setisValidLanguageCode(true);setBtnDisabled(false)} else {...}`

Reply:
could the `btnDisabled` be `!isValidLanguageCode` ?